src/wp-includes/formatting.php

```diff
@@ -5394,6 +5394,28 @@ function normalize_whitespace( $str ) {
  * @return string The processed string.
  */
 function wp_strip_all_tags( $text, $remove_breaks = false ) {
+	if ( is_null( $text ) ) {
+		return '';
+	}
+
+	if ( ! is_scalar( $text ) ) {
+		if ( WP_DEBUG ) {
+			trigger_error(
+				sprintf(
+					/* translators: 1: The function name, 2: The argument number, 3: The expected type, 4: The provided type. */
+					__( 'Warning: %1$s expects parameter %2$d to be a %3$s, %4$s given.' ),
+					__FUNCTION__,
+					'1',
```

Comment:
I'd like to suggest changing the message to use the parameter name, rather than it's position is now an implementation detail since PHP 8.0 introduced named arguments in function calls.

Also: why is `trigger_error()` being used here instead of `_doing_it_wrong()` ?

Reply:
> I'd like to suggest changing the message to use the parameter name, rather than it's position is now an implementation detail since PHP 8.0 introduced named arguments in function calls.

Work for me, the translator note will need to be updated too.

> Also: why is `trigger_error()` being used here instead of `_doing_it_wrong()` ?

I discussed this with @hellofromtonya a while back (failed to log it on the ticket, sorry) and figured that maintaining consistency with the pre PHP 8 error levels was for the best, thus `E_USER_WARNING` rather than a notice.

Reply:
We may need to add an (internal) comment to explain the reason for using `trigger_error()` as I fear someone will change it to `_doing_it_wrong()` sooner or later otherwise.

Reply:
I added a comment to explain the reason for using `trigger_error()` in [7f6ba4a](https://github.com/WordPress/wordpress-develop/pull/3132/commits/7f6ba4a8f8245ecbc954cc89330fe10491d8d554).